netcheck/netcheck.go

```diff
@@ -24,10 +26,13 @@ type Report struct {
 	UDP                   bool                     // UDP works
 	IPv6                  bool                     // IPv6 works
 	MappingVariesByDestIP opt.Bool                 // for IPv4
+	HairPinning           opt.Bool                 // for IPv4
 	DERPLatency           map[string]time.Duration // keyed by STUN host:port
 }
 
 func GetReport(ctx context.Context, logf logger.Logf) (*Report, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
Maybe add a comment explaining that this ctx is to ensure closeOnCtx is called on return.

Reply:
Done.